src/qt/paymentserver.cpp

```diff
@@ -770,7 +770,7 @@ void PaymentServer::handlePaymentACK(const QString& paymentACKMsg)
 
 bool PaymentServer::verifyNetwork(const payments::PaymentDetails& requestDetails)
 {
-    bool fVerified = requestDetails.network() == Params().NetworkIDString();
+    bool fVerified = requestDetails.genesis() == Params().hashGenesisBlock;
     if (!fVerified) {
         qWarning() << QString("PaymentServer::%1: Payment request network \"%2\" doesn't match client network \"%3\".")
             .arg(__func__)
```

Comment:
One line below that (github doesn't like me) `.network()` doesn't exist here anymore.